refgenconf/refgenconf.py

```diff
@@ -186,13 +318,71 @@ def _invert_genomes(self):
         return genomes
 
 
-def select_genome_config(filename, conf_env_vars=None):
+def _download_json(url):
+    """
+    Safely connects to the provided API endpoint and downloads the JSON formatted data
+
+    :param str url: server API endpoint
+    :return dict: served data
+    """
+    import json, requests
+    _LOGGER.debug("Downloading JSON data; querying URL: '{}'".format(url))
+    try:
+        server_resp = requests.get(url)
+    except Exception as e:
+        reason = "{}: {}".format(e.__class__.__name__, e)
+    else:
+        if server_resp.ok:
+            return json.loads(server_resp.content)
+        reason = "Response status: {}".format(server_resp.status_code)
+    _LOGGER.warning("Error querying '{}' -- {}".format(url, reason))
+
+
+def _download_url_to_file(url, filepath):
+    """
+    Download asset at given URL to given filepath.
+
+    :param str url: URL to download
+    :param str filepath: path to file to save download
+    """
+    with urllib.request.urlopen(url) as response, open(filepath, 'wb') as outf:
+        shutil.copyfileobj(response, outf)
+
+
+def _is_large_archive(size):
```

Comment:
make make the "large" a defaulted parameter so

```
def _is_large_archive(size, larger_than)
```

so it can be updated easily?